src/__tests__/Components.react.test.js

```diff
@@ -0,0 +1,156 @@
+  });
+
+  test('should render children correctly', async () => {
+    const { container } = render(
+      <Card
+        id="AAPL"
+        ticker="AAPL"
+        changes={1.2}
+        price="250"
+        changesPercentage="0.23"
+        companyName="Apple Inc."
+      />,
+      { wrapper: MemoryRouter },
+    );
+
+    const companyContainer = container.querySelector('[data-testid="company"]');
+    const cardDetails = container.querySelector('[data-testid="card-details"]');
+
+    expect(await within(container).findByText('AAPL')).toBeTruthy();
+    expect(companyContainer).toBeInTheDocument();
+    expect(companyContainer.children.length).toBe(2);
+    expect(await within(companyContainer).findByText('Apple Inc.')).toBeTruthy();
+    expect(cardDetails).toBeInTheDocument();
+    expect(cardDetails.children.length).toBe(3);
+    expect(await within(cardDetails).findByText('$250')).toBeTruthy();
+    expect(await within(cardDetails).findByText(1.2)).toBeTruthy();
+    expect(await within(cardDetails).findByText('0.23')).toBeTruthy();
+  });
+});
+
+describe('Details Card', () => {
+  test('should render component correctly', () => {
+    const { asFragment } = render(
+      <DetailsCard
+        first="2020-02-02"
+        second="Filling Date: 2020-02-05"
+        third="Accepted Date: 2020-02-05"
+      >
+        <span />
+      </DetailsCard>,
+    );
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('should render children correctly', async () => {
+    const { container } = render(
+      <DetailsCard
+        first="2020-02-02"
+        second="Filling Date: 2020-02-05"
+        third="Accepted Date: 2020-02-05"
+      >
+        <span />
+      </DetailsCard>,
+    );
+
+    const first = container.querySelector('[data-testid="first"]');
+    const second = container.querySelector('[data-testid="second"]');
+    const third = container.querySelector('[data-testid="third"]');
+
+    expect(await within(container).findByText('2020-02-02')).toBeTruthy();
+    expect(await within(container).findByText('Filling Date: 2020-02-05')).toBeTruthy();
+    expect(await within(container).findByText('Accepted Date: 2020-02-05')).toBeTruthy();
+    expect(first).toBeInTheDocument();
+    expect(second).toBeInTheDocument();
+    expect(third).toBeInTheDocument();
+  });
+});
+
+describe('Header', () => {
+  test('should render component correctly', () => {
+    const { asFragment } = render(<Header home />, { wrapper: MemoryRouter });
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('should render children correctly', async () => {
+    const { container } = render(<Header home />, { wrapper: MemoryRouter });
+
+    const nav = container.querySelector('[data-testid="nav"]');
+    const icons = container.querySelector('[data-testid="icons"]');
+
+    expect(nav.children.length).toBe(2);
+    expect(icons.children.length).toBe(2);
+    expect(await within(nav).findByText('Stock Metrics')).toBeTruthy();
+  });
+});
```

Comment:
- [ ] The test on rendering the Header component is failing. Kindly fix that

![image](https://user-images.githubusercontent.com/50941074/189502768-b612882a-e0b0-4d83-93fa-4090edb1a422.png)